src/tarantool.c

```diff
@@ -679,10 +689,22 @@ int get_spaceno_by_name(tarantool_connection *obj, zval *name) {
 		tarantool_throw_parsingexception("schema (space)");
 		return FAILURE;
 	}
-	space_no = tarantool_schema_get_sid_by_string(obj->schema,
-			Z_STRVAL_P(name), Z_STRLEN_P(name));
-	if (space_no == FAILURE)
-		THROW_EXC("No space '%s' defined", Z_STRVAL_P(name));
+
+	if (Z_TYPE_P(name) == IS_LONG) {
+		space_no = tarantool_schema_get_sid_by_number(obj->schema,
+							      Z_LVAL_P(name));
+		if (space_no == FAILURE) {
+			THROW_EXC("No space '%s' defined", Z_STRVAL_P(name));
```

Comment:
It is a long value, not a string. Causes a segfault in case of select using an unknown space id.

Reply:
Fixed.